var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -48,6 +48,14 @@ def determine_version(cls, exe):
         match = re.search(r'GNU M4\)?\s+(\S+)', output)
         return match.group(1) if match else None
 
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        # Inform autom4te if it wasn't built correctly (some external
+        # installations such as homebrew)
+        env.set('M4', self.prefix.bin.m4)
```

Comment:
Can you add a reference to [this page](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/autom4te-Invocation.html)? That's the only doc page I could find that reports:
```
By default, GNU M4 is found during configuration, but the environment variable M4 can be set to tell autom4te where to look.
```